gobblin-modules/gobblin-parquet/src/main/java/org/apache/gobblin/converter/parquet/JsonElementConversionFactory.java

```diff
@@ -0,0 +1,527 @@
+
+    public LongConverter(JsonSchema schema, boolean repeated, WorkUnitState state) {
+      super(schema, repeated, INT64, state);
+    }
+
+    @Override
+    LongValue convertField(JsonElement value) {
+      return new LongValue(value.getAsLong());
+    }
+
+    @Override
+    public Type schema() {
+      return this.schema;
+    }
+  }
+
+  public static class FloatConverter extends PrimitiveConverter {
+
+    public FloatConverter(JsonSchema schema, boolean repeated, WorkUnitState state) {
+      super(schema, repeated, PrimitiveTypeName.FLOAT, state);
+    }
+
+    @Override
+    FloatValue convertField(JsonElement value) {
+      return new FloatValue(value.getAsFloat());
+    }
+
+    @Override
+    public Type schema() {
+      return this.schema;
+    }
+  }
+
+  public static class DoubleConverter extends PrimitiveConverter {
+
+    public DoubleConverter(JsonSchema schema, boolean repeated, WorkUnitState state) {
+      super(schema, repeated, PrimitiveTypeName.DOUBLE, state);
+    }
+
+    @Override
+    DoubleValue convertField(JsonElement value) {
+      return new DoubleValue(value.getAsDouble());
+    }
+
+    @Override
+    public Type schema() {
+      return this.schema;
+    }
+  }
+
+  public static class BooleanConverter extends PrimitiveConverter {
+
+    public BooleanConverter(JsonSchema schema, boolean repeated, WorkUnitState state) {
+      super(schema, repeated, PrimitiveTypeName.BOOLEAN, state);
+    }
+
+    @Override
+    BooleanValue convertField(JsonElement value) {
+      return new BooleanValue(value.getAsBoolean());
+    }
+
+    @Override
+    public Type schema() {
+      return this.schema;
+    }
+  }
+
+  public static class StringConverter extends PrimitiveConverter {
+
+    public StringConverter(JsonSchema schema, boolean repeated, WorkUnitState state) {
+      super(schema, repeated, BINARY, state);
+      this.schema = buildSchema();
+    }
+
+    @Override
+    BinaryValue convertField(JsonElement value) {
+      return new BinaryValue(Binary.fromString(value.getAsString()));
+    }
+
+    private Type buildSchema() {
+      String columnName = this.jsonSchema.getColumnName();
+      if (this.repeated) {
+        return Types.repeated(BINARY).as(UTF8).named(columnName);
+      }
+      switch (this.jsonSchema.optionalOrRequired()) {
+        case OPTIONAL:
+          return Types.optional(BINARY).as(UTF8).named(columnName);
+        case REQUIRED:
+          return Types.required(BINARY).as(UTF8).named(columnName);
+        default:
+          throw new RuntimeException("Unsupported Repetition type");
+      }
+    }
+
+    @Override
+    public Type schema() {
+      return this.schema;
+    }
+  }
+
+  public static class ArrayConverter extends CollectionConverter {
+
+    public ArrayConverter(JsonSchema arraySchema, WorkUnitState state) {
+      super(arraySchema, arraySchema.getElementTypeUsingKey(ARRAY_ITEMS_KEY), state, true);
+      this.schema = buildSchema();
```

Comment:
Can move this line and the function `schema()` to `CollectionConverter`. Subclasses could just override `buildSchema()`